openedx/core/djangoapps/user_authn/cookies.py

```diff
@@ -158,9 +158,9 @@ def refresh_jwt_cookies(request, response, user):
     """
     Resets the JWT related cookies in the response for the given user.
     """
-    if user.is_authenticated and not user.is_anonymous:
-        cookie_settings = standard_cookie_settings(request)
-        _create_and_set_jwt_cookies(response, request, cookie_settings, user=user)
+    cookie_settings = standard_cookie_settings(request)
```

Comment:
I may need a mini-walkthrough of why this was needed, and why it can't be consistent with https://github.com/edx/edx-platform/blob/e4f935aab3f1b9621758131520cacc74158ab99d/openedx/core/djangoapps/user_authn/cookies.py#L143 .
Additionally, if it really needs to be inconsistent, we should add a `WARNING` comment that this will set the cookies regardless of the situation.

Reply:
It is consistent, we are just performing the check for the anonymous user a layer up in the view.